packages/@tailwindcss-cli/src/commands/build/index.ts

```diff
@@ -212,8 +213,9 @@ export async function handle(args: Result<ReturnType<typeof options>>) {
             // Re-scan the directory to get the new `candidates`
             scanner = new Scanner({
               detectSources: { base },
-              sources: compiler.globs.map((pattern) => ({
-                base: inputBasePath, // Globs are relative to the input.css file
+              sources: compiler.globs.map(({ base, pattern }) => ({
+                // Globs are relative to the input.css file
+                base: base ? path.dirname(path.resolve(inputBasePath, base)) : inputBasePath,
```

Comment:
This makes me wonder if we shouldn't call this something else? We can't resolve the base inside tailwindcss core (if we do, we need to make it aware of paths etc. which feels wrong if it doesn't do IO?) so instead, we have to translate the paths that are relative to the input CSS file here to the proper base dirname.

Perhaps it should be caled `origin` instead of `base`? That would also make it clearer why `origin` can be `null` (i.e it's origin is the input CSS file)

cc @thecrypticace @RobinMalfait @adamwathan

Reply:
I like origin if it’s a full file name and not the directory, is that the case? The full file path feels more correct to me with that name (“this is the file this source came from”) if it doesn’t introduce any other problems.

Reply:
> I like origin if it’s a full file name and not the directory, is that the case? 

Yeah that's the case, it's basically just keeping track of whatever string is after `@plugin` or `@config` right now.